### PR TITLE
AO3-6858 Add puma library

### PR DESCRIPTION
**Pull request by zz9pzza**
# Pull Request Checklist

* [ ] Have you read ["How to write the perfect pull request"](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)?
* [ ] Have you read the [contributing guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [ ] Have you added [tests for any changed functionality](https://github.com/otwcode/otwarchive/wiki/Automated-Testing)?
* [ ] Have you added the [Jira](https://otwarchive.atlassian.net) issue number
  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
* [ ] Do you have fewer than 5 pull requests already open? If not, please wait
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6858

## Purpose

Adds puma install, it does not start to use it 

## Testing Instructions

This change does not have any functional changes. It allows us to migrate to puma 

## References

Are there other relevant issues/pull requests/mailing list discussions?

## Credit

What name and pronouns should we use to credit you in the [Archive of Our Own's Release Notes](https://archiveofourown.org/admin_posts?tag=1)?

james_

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -634,7 +636,7 @@ GEM
     zeitwerk (2.6.18)
 
 PLATFORMS
-  ruby
+  x86_64-linux
```
Comment (Bilka2) [inline Gemfile.lock:639]: Can you revert this change? I tried it locally and bundler doesn't complain.

**Comment by Bilka2** [comment]
In my Docker install, Rails uses Puma automatically instead of WEBrick. Is that wanted?

**Comment by zz9pzza** [comment]
> In my Docker install for development, Rails uses Puma automatically instead of WEBrick. Is that wanted?

Its not unexpected, I am happy with that, it will not effect production

**Review by brianjaustin — APPROVED** [review]
